transaction-service/src/test/kotlin/ua/com/foxminded/restClient/NatsControllerTest.kt

```diff
@@ -0,0 +1,137 @@
+package ua.com.foxminded.restClient
+
+import com.google.protobuf.Timestamp
+import io.nats.client.Connection
+import io.nats.client.Message
+import io.nats.client.Nats
+import java.time.LocalDateTime
+import java.time.ZoneId
+import java.util.Currency
+import java.util.UUID
+import java.util.concurrent.CountDownLatch
+import kotlin.test.assertEquals
+import org.junit.jupiter.api.AfterAll
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+import org.mockito.Mockito
+import org.mockito.kotlin.any
+import org.mockito.kotlin.eq
+import org.springframework.data.domain.Pageable
+import org.testcontainers.containers.GenericContainer
+import org.testcontainers.junit.jupiter.Testcontainers
+import reactor.core.publisher.Flux
+import ua.com.foxminded.restClient.dto.TransactionDto
+import ua.com.foxminded.restClient.enums.Direction
+import ua.com.foxminded.restClient.nats.NatsController
+import ua.com.foxminded.restClient.service.CurrencyExchangeService
+import ua.com.foxminded.restClient.service.TransactionService
+
+@Testcontainers
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class NatsControllerTest {
+
+    val testDtos: List<TransactionDto> = getListTransactions()
+
+    val transactionService = Mockito.mock(TransactionService::class.java)
+
+    val exchangeService = Mockito.mock(CurrencyExchangeService::class.java)
+
+    lateinit var natsUrl: String
+
+    lateinit var natsConnection: Connection
+
+    lateinit var natsController: NatsController
+
+    lateinit var nats: GenericContainer<*>
+
+    @BeforeAll
+    fun setUp() {
+        nats = GenericContainer("nats:latest").withExposedPorts(NATS_PORT, NATS_MGMT_PORT)
+        nats.start()
+        natsUrl = "nats://%s:%d".format(nats.host, nats.getMappedPort(NATS_PORT))
+        natsConnection = Nats.connect(natsUrl)
+        natsController = NatsController(transactionService, exchangeService, natsConnection)
+    }
+
+    @Test
+    fun nats_shouldReturnFluxTransactions_whenInputCorrect() {
```

Comment:
you can rename with 
```kotlin
fun `nats should return flux ...`() {
```

It's much more easier to read

Reply:
okay.